wsagent/che-core-workspace-activity-server/src/main/java/org/eclipse/che/workspace/activity/WorkspaceActivityNotifier.java

```diff
@@ -53,16 +53,24 @@ public WorkspaceActivityNotifier(
   /**
    * Notify workspace master about activity in this workspace.
    *
-   * <p>After last notification, any consecutive activities that come within specific amount of time
-   * - {@code threshold}, will not notify immediately, but trigger notification in scheduler method
-   * {@link WorkspaceActivityNotifier#scheduleActivityNotification}
+   * <p>There is a {@code threshold} that limits request sending frequency - if no activity has
+   * occurred within the duration of threshold, then the request will be sent immediately (in a
+   * separate thread, to prevent it's possible long execution from blocking of the Tomcat thread,
+   * from which it may be called).
+   *
+   * <p>In case there was any recent activity within threshold time, any consecutive activities will
+   * not send request immediately, but trigger it's sending in scheduler method {@link
+   * WorkspaceActivityNotifier#scheduleActivityNotification}
    */
   public void onActivity() {
     long currentTime = System.currentTimeMillis();
     if (currentTime < (lastUpdateTime + threshold)) {
       activeDuringThreshold.set(true);
     } else {
-      notifyActivity();
+      Thread activityRequestThread = new Thread(this::notifyActivity);
+      activityRequestThread.setDaemon(true);
```

Comment:
give some meaningful name to this thread.

Reply:
ok